l10n_nl_intrastat/models/l10n_nl_intrastat.py

```diff
@@ -0,0 +1,220 @@
+        """
+        Collect the data lines for the given report.
+        Unlink any existing lines first.
+        """
+        self.ensure_one()
+        # Other models:
+        invoice_model = self.env['account.invoice']
+        invoice_line_model = self.env['account.invoice.line']
+
+        # Check whether all configuration done to generate report
+        self._check_generate_lines()
+
+        # We need at least these variables to calculate
+        if not self.date_from or not self.date_to or not self.company_id:
+            return
+
+        # Define search for invoices for period and company:
+        company_obj = self.company_id  # simplify access
+        invoice_domain = [
+            ('type', 'in', ('out_invoice', 'out_refund')),
+            ('date_invoice', '>=', self.date_from),
+            ('date_invoice', '<=', self.date_to),
+            ('state', 'in', ('open', 'paid')),
+            ('company_id', '=', self.company_id.id),
+        ]
+        # Signal invoices without a country
+        # (Should not happen, as country_id on parter is set to required)
+        invalid_invoices = invoice_model.search_read(
+            domain=invoice_domain + [('partner_id.country_id', '=', False)],
+            fields=['partner_id']
+        )
+        if invalid_invoices:
+            raise Warning(
+                _(
+                    "Missing country on the invoice addresses of the"
+                    " following partners:\n%s"
+                ) % "\n".join(
+                    inv['partner_id'][1] for inv in invalid_invoices)
+            )
+        # Search invoices that need intrastat reporting:
+        invoice_records = invoice_model.search(
+            invoice_domain + [
+                ('partner_id.country_id.intrastat', '=', True),
+                ('partner_id.country_id.id', '!=', company_obj.country_id.id),
+            ]
+        )
+        invoice_ids = [inv['id'] for inv in invoice_records]
+        invoice_line_records = invoice_line_model.search(
+            [('invoice_id', 'in', invoice_ids)])
+
+        # Gather amounts from invoice lines
+        total_amount = 0.0
+        partner_amounts_map = {}
+        for line in invoice_line_records:
+            # Ignore invoiceline if taxes should not be included in intrastat:
+            if any(tax.exclude_from_intrastat_if_present for tax in
+                    line.invoice_line_tax_ids):
+                continue
+            # Report is per commercial partner:
+            commercial_partner_id = (
+                line.invoice_id.partner_id.commercial_partner_id.id)
+            if commercial_partner_id not in partner_amounts_map:
+                partner_amounts_map[commercial_partner_id] = {
+                    'amount_product': 0.0,
+                    'amount_service': 0.0,
+                }
+            amounts = partner_amounts_map[commercial_partner_id]
+            # Determine product or service:
+            if (line.product_id and (
```

Comment:
that's up to the person doing the refactoring. With the same argument, you could want to name all default parameters of a function because somebody could change the defaults